pkg/scheduler/controller/backupbucket/scheduler_reconciler.go

```diff
@@ -0,0 +1,229 @@
+	key, err := cache.MetaNamespaceKeyFunc(obj)
+	if err != nil {
+		logger.Logger.Errorf("Couldn't get key for object %+v: %v", obj, err)
+		return
+	}
+
+	newBackupBucket := obj.(*gardencorev1alpha1.BackupBucket)
+
+	if newBackupBucket.DeletionTimestamp != nil {
+		logger.Logger.Infof("Ignoring backupBucket '%s' because it has been marked for deletion", newBackupBucket.Name)
+		c.backupBucketQueue.Forget(key)
+		return
+	}
+
+	c.backupBucketQueue.Add(key)
+}
+
+func (c *SchedulerController) backupBucketUpdate(oldObj, newObj interface{}) {
+	c.backupBucketAdd(newObj)
+}
+
+// reconciler implements the reconcile.Reconcile interface for backupBucket scheduler.
+type reconciler struct {
+	ctx      context.Context
+	client   client.Client
+	recorder record.EventRecorder
+	logger   *logrus.Entry
+}
+
+// newReconciler returns the new backupBucker reconciler.
+func newReconciler(ctx context.Context, gardenClient client.Client, recorder record.EventRecorder) reconcile.Reconciler {
+	return &reconciler{
+		ctx:      ctx,
+		client:   gardenClient,
+		recorder: recorder,
+		logger:   logger.NewFieldLogger(logger.Logger, "scheduler", "backupbucket"),
+	}
+}
+func (r *reconciler) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	bb := &gardencorev1alpha1.BackupBucket{}
+	if err := r.client.Get(r.ctx, request.NamespacedName, bb); err != nil {
+		if apierrors.IsNotFound(err) {
+			r.logger.Debugf("[SCHEDULER BACKUPBUCKET RECONCILE] %s - skipping because BackupBucket has been deleted", request.NamespacedName)
+			return reconcile.Result{}, nil
+		}
+		r.logger.Infof("[SCHEDULER BACKUPBUCKET RECONCILE] %s - unable to retrieve object from store: %v", request.NamespacedName, err)
+		return reconcile.Result{}, err
+	}
+
+	return reconcile.Result{}, r.scheduleBackupBucket(bb)
+}
+
+type executeSchedulingRequest = func(context.Context, *gardencorev1alpha1.BackupBucket) error
+
+func (r *reconciler) scheduleBackupBucket(obj *gardencorev1alpha1.BackupBucket) error {
+	var (
+		backupBucket    = obj.DeepCopy()
+		schedulerLogger = r.logger.WithField("backupbucket", backupBucket.Name)
+	)
+
+	if backupBucket.Spec.Seed != nil {
+		// If the BackupBucket manifest already specifies a desired Seed cluster,
+		// we should check its availability. If its not available we will try reschedule it again.
+		schedulerLogger.Infof("BackupBucket is already scheduled on seed %s", *backupBucket.Spec.Seed)
+		seed := &gardenv1beta1.Seed{}
+		if err := r.client.Get(r.ctx, kutil.Key(*backupBucket.Spec.Seed), seed); err != nil {
+			return err
+		}
+
+		if cond := gardencorev1alpha1helper.GetCondition(seed.Status.Conditions, gardenv1beta1.SeedAvailable); cond != nil && cond.Status != gardencorev1alpha1.ConditionUnknown {
+			schedulerLogger.Infof("Seed %s is available, ignoring further reconciliation.", *backupBucket.Spec.Seed)
+		}
+		schedulerLogger.Infof("Seed %s is not available, we will schedule it on another seed", *backupBucket.Spec.Seed)
```

Comment:
Well, that's dangerous, let's not do this for now. We should be very sure before we schedule it on another seed. I suggest to return here, for now.

Reply:
ok. As discussed i'll make `seed` field immutable and return it. Basiaclly while  deploying backupBucket in seed_control.go i'll assign seed. So, scheduler will always find seed assigned and do nothing.